### PR TITLE
Fix _detect_impacts back propagation

### PR DESCRIPTION
**Pull request by KatKiker**

### CODE REVIEW

File: tests/test_impacts.py [hunk 1]
```diff
@@ -69,4 +71,17 @@ def test_calculate_impacts_benchmark(benchmark, processes):
         seed=42  # This allows us to predict exact number of impactors empirically
     )
     assert len(variants) == 200, "Should have 200 variants"
-    assert len(impacts) == 0, "Should have exactly 0 impactors"
\ No newline at end of file
+    assert len(impacts) == 0, "Should have exactly 0 impactors"
+
+
+def test_detect_impacts():
```
Comment (akoumjian) [inline tests/test_impacts.py:77]:
```suggestion
def test_detect_impacts_time_direction():
```

File: src/adam_core/propagator/adam_assist.py [hunk 2]
```diff
@@ -292,11 +292,16 @@ def _detect_impacts(
...
 
+        backward_propagation = num_days < 0
+
         # Set the simulation time, relative to the jd_ref
         start_tdb_time = orbits.coordinates.time.jd().to_numpy()[0]
         start_tdb_time = start_tdb_time - ephem.jd_ref
         sim.t = start_tdb_time
 
+        if backward_propagation:
+            sim.dt = -0.01
```
Comment (akoumjian) [inline src/adam_core/propagator/adam_assist.py:303]: Can we change this to simply multiply the existing sim.dt by -1?
```suggestion
            sim.dt = sim.dt * -1
```

**Review by akoumjian — CHANGES_REQUESTED** [review]
Couple small changes, then approved.

**Comment by KatKiker** [comment]
Updated!

**Review by akoumjian — APPROVED** [review]

**Pull request merged**